openstack/blockstorage/v2/volumes/requests.go

```diff
@@ -0,0 +1,418 @@
+package volumes
+
+import (
+	"fmt"
+
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/pagination"
+)
+
+// CreateOptsBuilder allows extensions to add additional parameters to the
+// Create request.
+type CreateOptsBuilder interface {
+	ToVolumeCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts contains options for creating a Volume. This object is passed to
+// the volumes.Create function. For more information about these parameters,
+// see the Volume object.
+type CreateOpts struct {
+	// The availability zone [OPTIONAL]
+	Availability string
+	// he volume description [OPTIONAL]
+	Description string
+	// One or more metadata key and value pairs to associate with the volume [OPTIONAL]
+	Metadata map[string]string
+	// The volume name [OPTIONAL]
+	Name string
+	// The size of the volume, in gibibytes (GiB) [REQUIRED]
+	Size int
+	// the ID of the existing volume snapshot [OPTIONAL]
+	SnapshotID string
+	// the ID of the existing volume [OPTIONAL]
+	SourceVolID string
+	// The ID of the image from which you want to create the volume.
+	// Required to create a bootable volume.
+	ImageID string
+	// The associated volume type [OPTIONAL]
+	VolumeType string
+	// The unique identifier of the tenant or account [OPTIONAL]
+	TenantID string
```

Comment:
I may be mistaken, but I don't believe TenantID is a valid create parameter. The Tenant ID is used in the URL to make the request, but AFAIK, it's not a parameter that can be set during creation.